eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -171,9 +171,18 @@ case object GetRoutingState
 
 case class RoutingState(channels: Iterable[PublicChannel], nodes: Iterable[NodeAnnouncement])
 
-case class Stash(updates: Map[ChannelUpdate, Set[ActorRef]], nodes: Map[NodeAnnouncement, Set[ActorRef]])
+// @formatter:off
+sealed trait GossipOrigin {
+  val actor: ActorRef
+}
+/** Gossip that we received from a remote peer. */
+case class RemoteGossip(actor: ActorRef) extends GossipOrigin
+/** Gossip that was generated by our node. */
+case class LocalGossip(actor: ActorRef) extends GossipOrigin
```

Comment:
Why not:

 ```scala
case object LocalGossip extends GossipOrigin
```

Reply:
I hesitated, I figured it would be better to keep the reference of the (local) sender in case we wanted to send it an ACK or something, but it's probably YAGNI so we can remove it and re-add it later if we need it.